lib/jmespath/nodes/slice.rb

```diff
@@ -45,7 +46,7 @@ def adjust_slice(length, start, stop, step)
         if step.nil?
           step = 1
         elsif step == 0
-          raise Errors::RuntimeError, 'slice step cannot be 0'
+          return runtime_error('slice step cannot be 0')
```

Comment:
That is an excellent point. I'll definitely move the check to the constructor.
As for the error, all the other implementations seem to throw a `invalid-value` error for this, not sure why the ruby version error is called `runtime` instead. I'll keep the error type the same, to make drop in replacement easier.